utils/java/src/main/java/com/logicalclocks/utils/MainClass.java

```diff
@@ -116,7 +116,17 @@ public static void main(String[] args) throws Exception {
     LOGGER.info("Hsfs utils write options: {}", writeOptions);
 
     if (op.equals("offline_fg_materialization") || op.equals("offline_fg_backfill")) {
-      SparkEngine.getInstance().streamToHudiTable(streamFeatureGroup, writeOptions);
+      SparkEngine engine = SparkEngine.getInstance();
+      boolean suceeded = false;
+      try {
+        engine.streamToHudiTable(streamFeatureGroup, writeOptions);
+        suceeded = true;
+      } finally {
+        LOGGER.info("Closing spark session...");
+        engine.closeSparkSession();
```

Comment:
I would move this code outside the if statement considering that if an exception happens, the job shuts down correctly. 
On the other hand, if by any chance the op is not one of those 2s, the if and the finally will be skipped entirely not closing the session.